node/primitives/src/approval.rs

```diff
@@ -86,6 +257,51 @@ pub struct AssignmentCert {
 	pub vrf: (VRFOutput, VRFProof),
 }
 
+/// A certification of assignment.
+#[derive(Debug, Clone, Encode, Decode, PartialEq, Eq)]
+pub struct AssignmentCertV2 {
+	/// The criterion which is claimed to be met by this cert.
+	pub kind: AssignmentCertKindV2,
+	/// The VRF showing the criterion is met.
+	pub vrf: (VRFOutput, VRFProof),
+}
```

Comment:
I'd expect `AssignmentCertV2` gains a `claimed_cores: Vec<u32>` field, which then both the delay and modulo variant hash into their extra data. 

In my original draft this "story" data could be pushed either into the extra data or into the VRF input, but I did little with it there. https://github.com/paritytech/polkadot/pull/1558/files